docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -33,203 +41,171 @@ To try out Camunda 8 or develop against it, consider signing up for our [SaaS of
 
 For the simplicity of this guide, certain best practices will be provided with links to additional documents, enabling you to explore the topic in more detail.
 
-:::warning
+:::warning Cost management
 Following this guide will incur costs on your Cloud provider account, namely for the managed Kubernetes service, running Kubernetes nodes in EC2, Elastic Block Storage (EBS), and Route53. More information can be found on [AWS](https://aws.amazon.com/eks/pricing/) and their [pricing calculator](https://calculator.aws/#/) as the total cost varies per region.
 :::
 
-## Outcome
+<!-- TODO : add architecture diagram --->
+
+### Outcome
 
 Following this tutorial and steps will result in:
 
 - An Amazon EKS Kubernetes cluster running the latest Kubernetes version with four nodes ready for Camunda 8 installation.
 - The [EBS CSI driver](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html) is installed and configured, which is used by the Camunda 8 Helm chart to create [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
 - A [managed Aurora PostgreSQL 15.8](https://aws.amazon.com/rds/postgresql/) instance to be used by the Camunda 8 components.
+- An OpenSearch domain created and configured for use with the Camunda 8 platform, leveraging the capabilities of [AWS OpenSearch Service](https://aws.amazon.com/opensearch-service/).
 
-## Installing Amazon EKS cluster with Terraform
+## 1. Provisioning the Complete Infrastructure for Camunda 8 on AWS
 
-### Terraform prerequsites
-
-1. Create an empty folder to place your Terraform files in.
-2. Create a `config.tf` with the following setup:
-
-```hcl
-terraform {
-  backend "local" {
-    path = "terraform.tfstate"
-  }
-
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = "~> 5.69"
-    }
-  }
-}
+### Terraform Prerequisites
 
-provider "aws" {
-  region     = "eu-central-1"
-}
-```
+To manage the infrastructure for Camunda 8 on AWS using Terraform, we need to set up Terraform's backend to store the state file remotely in an S3 bucket. This ensures secure and persistent storage of the state file.
```

Comment:
it sounds like a hard requirement?

While I agree that it makes sense, nothing speaks against any other backend that Terraform offers either.

Reply:
it's also more or less introducing a hen and egg problem of `wait a minute, my s3 bucket isn't in IaC but I need one for IaC` 🤯